src/main/scala/com.snowplowanalytics.snowplow.micro/Configuration.scala

```diff
@@ -138,18 +139,38 @@ object Configuration {
       .flatMap(resolverConfig => buildIgluResources(resolverConfig, maxJsonDepth))
   }
 
-  private def loadEnrichmentConfig(igluClient: IgluCirceClient[IO]): EitherT[IO, String, List[EnrichmentConf]] = {
-    Option(getClass.getResource("/enrichments")) match {
-      case Some(definedEnrichments) =>
-        val path = Paths.get(definedEnrichments.toURI)
-        for {
-          asJson <- loadEnrichmentsAsSDD(path, igluClient, fileType = ".json")
-          asHocon <- loadEnrichmentsAsSDD(path, igluClient, fileType = ".hocon")
-          asJSScripts <- loadJSScripts(path)
-        } yield asJson ::: asHocon ::: asJSScripts
-      case None =>
-        EitherT.rightT[IO, String](List.empty)
-    }
+  private def loadEnrichmentConfig(igluClient: IgluCirceClient[IO], enableYauaa: Boolean): EitherT[IO, String, List[EnrichmentConf]] = {
+    for {
+      userEnrichments <- Option(getClass.getResource("/enrichments")) match {
+        case Some(definedEnrichments) =>
+          val path = Paths.get(definedEnrichments.toURI)
+          for {
+            asJson <- loadEnrichmentsAsSDD(path, igluClient, fileType = ".json")
+            asHocon <- loadEnrichmentsAsSDD(path, igluClient, fileType = ".hocon")
+            asJSScripts <- loadJSScripts(path)
+          } yield asJson ::: asHocon ::: asJSScripts
+        case None =>
+          EitherT.rightT[IO, String](List.empty[EnrichmentConf])
+      }
+      enrichmentsWithYauaa = if (enableYauaa) {
+        val hasYauaa = userEnrichments.exists { conf =>
+          conf.schemaKey.vendor == "com.snowplowanalytics.snowplow.enrichments" &&
+          conf.schemaKey.name == "yauaa_enrichment_config"
+        }
+
+        if (hasYauaa) {
+          userEnrichments
+        } else {
+          val yauaaEnrichment = EnrichmentConf.YauaaConf(
+            schemaKey = SchemaKey("com.snowplowanalytics.snowplow.enrichments", "yauaa_enrichment_config", "jsonschema", SchemaVer.Full(1, 0, 0)),
+            cacheSize = None
+          )
+          userEnrichments :+ yauaaEnrichment
+        }
+      } else {
+        userEnrichments
+      }
+    } yield enrichmentsWithYauaa
   }
```

Comment:
I think that this function could be made easier to read by splitting it. We use a `for {} yield` whereas only one `flatMap` happens.

One alternative would be to create a function e.g.

```
def maybeAddYauaa(userEnrichments: List[EnrichmentConf], enableYauaa: Boolean): List[EnrichmentConf]
```

And then to do something like:

```
val userEnrichments = ...
userEnrichments.map(maybeAddYauaa(_, enableYauaa)
```